File: 2.py
def core(value: list, isRalse: list, index: int, k: int) -> int:
    result = 0
    for i in range(k):
        if (isRalse[index + i] == False):
            result += value[index + 1]#这里马虎把i打成1，导致下边的测试用例不通过
    return result


import numpy as np

if __name__ == '__main__':
    # n,k = map(int,input().strip().split(" "))
    # value = [int(i) for i in input().strip().split(" ")]
    # isRalse = [int(i)==1 for i in input().strip().split(" ")]
    n, k = 10, 5
    value = [int(i) for i in "6481 6127 4477 5436 7356 3137 1076 7182 8147 835".split(" ")]
    isRalse = [int(i) == 1 for i in "1 0 1 0 1 1 0 0 0 1".strip().split(" ")]
    rightSum = np.dot(np.array([int(i) for i in "6481 6127 4477 5436 7356 3137 1076 7182 8147 835".split(" ")]) , np.array(
        [int(i) for i in "1 0 1 0 1 1 0 0 0 1".strip().split(" ")]).T)
    sum = 0
    maxNum = 0
    for i in range(n):
        if (isRalse[i]):
            sum += value[i]
        if (i >= n - k + 1):
            continue
        temp = core(value, isRalse, i, k)
        if (temp > maxNum):
            maxNum = temp
    print("%d" % (sum + maxNum))

File: 2.cpp
//
// Created by tmy on 19-9-1.
//
#include <iostream>
#include <vector>
void initValue(std::vector<int >& value){
    for(auto Iter=value.begin();Iter!=value.end();Iter++){
//        std::scanf("%d",Iter);
        std::cin>>(*Iter);
    }
}
void initIsRalse(std::vector<bool >& isRalse){
    for(auto Iter=isRalse.begin();Iter!=isRalse.end();Iter++){
        int temp;
//        std::scanf("%d",&temp);
        std::cin>>temp;
        *Iter = (temp==1);
    }
}
int core(std::vector<int >& value,std::vector<bool >& isRalse,int index,int k){
    int result=0;
    for(int i=0;i<k;i++){
        if(!isRalse[index+i]){
            result += value[index+i];
        }
    }
    return result;
}
int main(){
    int n,k;
    std::cin>>n;
    std::cin>>k;
    std::vector<int > value(n);
    std::vector<bool > isRalse(n);
    initValue(value);
    initIsRalse(isRalse);
    int sum = 0;
    int max = 0;
    for(int i=0;i<n;i++){
        if(isRalse[i]){sum += value[i];}
        if(i >= n-k+1){ continue;}
        int temp = core(value,isRalse,i,k);
        if(temp > max){
            max = temp;
        }
    }
    std::cout<<(sum+max);
    return 0;
}
